tests/e2e/system_test/test_mcg_replication_with_disruptions.py

```diff
@@ -0,0 +1,243 @@
+            test_directory_setup.origin_dir,
+            mcg_obj=mcg_obj_session,
+            amount=5,
+            pattern="first-write-",
+        )
+        logger.info(f"Written objects: {written_random_objects}")
+
+        compare_bucket_object_list(
+            mcg_obj_session, source_bucket_name, target_bucket_name
+        )
+        logger.info("Uni-directional bucket replication working as expected")
+
+        # change from uni-directional to bi-directional replication policy
+        logger.info("Changing the replication policy from uni to bi-directional!")
+        bi_replication_policy_dict = {
+            "spec": {
+                "additionalConfig": {
+                    "replicationPolicy": json.dumps(
+                        [
+                            {
+                                "rule_id": "basic-replication-rule-2",
+                                "destination_bucket": source_bucket_name,
+                            }
+                        ]
+                    )
+                }
+            }
+        }
+        OCP(
+            namespace=config.ENV_DATA["cluster_namespace"],
+            kind="obc",
+            resource_name=target_bucket_name,
+        ).patch(params=json.dumps(bi_replication_policy_dict), format_type="merge")
+        logger.info(
+            "Patch ran successfully! Changed the replication policy from uni to bi directional"
+        )
+
+        # write objects to the second bucket and see if it's replicated on the other
+        logger.info("checking if bi-directional replication works!!")
+        written_random_objects = write_random_test_objects_to_bucket(
+            awscli_pod_session,
+            target_bucket_name,
+            test_directory_setup.origin_dir,
+            mcg_obj=mcg_obj_session,
+            amount=3,
+            pattern="second-write-",
+        )
+        logger.info(f"Written objects: {written_random_objects}")
+        compare_bucket_object_list(
+            mcg_obj_session, source_bucket_name, target_bucket_name
+        )
+        logger.info("Bi directional bucket replication working as expected")
+
+        # delete all the s3-compatible namespace buckets objects and then recover it from other namespace bucket on
+        # write
+        logger.info(
+            "checking replication when one of the bucket's objects are deleted!!"
+        )
+        rgw_objects = [
+            obj.key
+            for obj in mcg_obj_session.s3_list_all_objects_in_bucket(target_bucket_name)
+        ]
+        try:
+            for obj in rgw_objects:
+                s3_delete_object(
+                    mcg_obj_session, bucketname=target_bucket_name, object_key=obj
+                )
+        except Exception as e:
+            logger.error(f"[Error] while deleting objects: {e}")
+        obj_after_deletion = [
+            obj.key
+            for obj in mcg_obj_session.s3_list_all_objects_in_bucket(target_bucket_name)
+        ]
```

Comment:
The length stays the same regardless of the processing, I don't think it is needed

Reply:
what do you mean by length remains the same? objects are restored back only after the next write IIUC. so length should be 0 after the deletion :?

Reply:
`len(mcg_obj_session.s3_list_all_objects_in_bucket(target_bucket_name)) == len(obj_after_deletion)`

Reply:
okay